index.js

```diff
@@ -9,38 +13,42 @@ class DuplicateCompatError extends Error {
   }
 }
 
-function load() {
-  // Recursively load one or more directories passed as arguments.
+/**
+ * Recursively load one or more directories passed as arguments.
+ *
+ * @param {string[]} dirs The directories to load
+ * @returns {object} All of the browser compatibility data
+ */
+function load(...dirs) {
   let dir,
     result = {};
 
-  function processFilename(fn) {
-    const fp = path.join(dir, fn);
-    let extra;
+  for (dir of dirs) {
+    dir = path.isAbsolute(dir) ? dir : path.resolve(__dirname, dir);
+    const files = fs.readdirSync(dir);
+    for (const fn of files) {
+      const fp = path.join(dir, fn);
+      let extra;
```

Comment:
The problem is outside of the diff so I'll have to comment here. The recursive `extra = load(fp)` means that we'll create a new `result` dictionary and merge multiple intermediate dictionaries, one for each directory nesting level. It's not a problem in practice, but if we're going to refactor this to be *nice* I'd like to change it.

What I'd suggest is a helper function for the recursion, and that `result` lives in the outer scope, just one. Every time we find a JSON file we extend that single object. WDYT?

Reply:
Ping @queengooborg

Reply:
I overhauled this to utilize `fdir` for directory walking, the same dependency used in `yari` now.  I then ran a release build before and after this change and compared the results with `assert.deepEqual()` to make sure our output has not changed, which it hasn't!